tests/system/projects/test_project.py

```diff
@@ -1729,6 +1729,20 @@ def test_notifications_on_workflow_start_and_completion(self):
             len(notifications) == 2
         ), f"Expected 2 notifications, got {len(notifications)}"
 
+    def test_set_and_run_remote_workflow(self):
```

Comment:
You need to create the project without copying the files.
1. Use `new_project` without source and then `project.set_source`
2. Verify that when using `kfp` engine this files with proper error message such as "File not found"